src/commands/synthetics/tunnel/tunnel.ts

```diff
@@ -34,6 +38,51 @@ export interface TunnelReporter {
   warn(message: string): void
 }
 
+export interface TunnelFirewallRules {
+  allowedSubnetsByFamily: typeof RESERVED_ADDRESS_BLOCKS
+  blockedSubnetsByFamily: typeof RESERVED_ADDRESS_BLOCKS
+}
+
+export const getTunnelFirewallRules = (config: RunTestsCommandConfig): TunnelFirewallRules => {
+  const {allowedIPRanges, blockedIPRanges, enableDefaultBlockedIPRanges} = config
+  const firewallRules: TunnelFirewallRules = {
+    allowedSubnetsByFamily: {
+      [IpFamily.v4]: [],
+      [IpFamily.v6]: [],
+    },
+    blockedSubnetsByFamily: {
+      [IpFamily.v4]: [],
+      [IpFamily.v6]: [],
+    },
+  }
+
+  const hasDefinedRanges = (firewallRanges: typeof RESERVED_ADDRESS_BLOCKS) =>
+    [IpFamily.v4, IpFamily.v6].some((family) => (firewallRanges[family] || []).length !== 0)
+
+  if (enableDefaultBlockedIPRanges) {
+    firewallRules.blockedSubnetsByFamily[IpFamily.v4].push(...(RESERVED_ADDRESS_BLOCKS[IpFamily.v4] || []))
+    firewallRules.blockedSubnetsByFamily[IpFamily.v6].push(...(RESERVED_ADDRESS_BLOCKS[IpFamily.v6] || []))
+  }
+
+  if (hasDefinedRanges(allowedIPRanges) || hasDefinedRanges(blockedIPRanges) || enableDefaultBlockedIPRanges) {
+    firewallRules.allowedSubnetsByFamily[IpFamily.v4].push(...(allowedIPRanges[IpFamily.v4] || []))
+    firewallRules.allowedSubnetsByFamily[IpFamily.v6].push(...(allowedIPRanges[IpFamily.v6] || []))
+    firewallRules.blockedSubnetsByFamily[IpFamily.v4].push(...(blockedIPRanges[IpFamily.v4] || []))
+    firewallRules.blockedSubnetsByFamily[IpFamily.v6].push(...(blockedIPRanges[IpFamily.v6] || []))
+  }
+
+  return firewallRules
+}
+
+const getIPFamily = (ipAddress: string): 4 | 6 => {
+  const ipFamily = isIP(ipAddress)
+  if (ipFamily === 4 || ipFamily === 6) {
+    return ipFamily
+  } else {
+    throw new Error('invalid IP address')
+  }
+}
```

Comment:
Not sure about that, should we use `IpFamily`?

```suggestion
const getIPFamily = (ipAddress: string): IpFamily.v4 | IpFamily.v6 => {
  const ipFamily = isIP(ipAddress)
  if (ipFamily === IpFamily.v4 || ipFamily === IpFamily.v6) {
    return ipFamily
  } else {
    throw new Error('invalid IP address')
  }
}
```